src/test/java/com/softdevsix/api/domain/report/ReportTest.java

```diff
@@ -0,0 +1,51 @@
+package com.softdevsix.api.domain.report;
+
+import com.softdevsix.api.exceptions.ReportReaderException;
+import com.softdevsix.api.services.report.IReportReader;
+import com.softdevsix.api.services.report.JsonReportReader;
+import org.junit.jupiter.api.Test;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+
+class ReportTest {
+    public static String readFile(String path) {
+        try (InputStream fileInputStream = Thread.currentThread().getContextClassLoader().getResourceAsStream(path)) {
+            StringBuilder file = new StringBuilder();
+            BufferedReader reader = new BufferedReader(new InputStreamReader(fileInputStream));
+            String line;
+            while ((line = reader.readLine()) != null) {
+                file.append(line).append("\n");
+            }
+            return file.toString();
+
+        } catch (IOException e) {
+            fail(e.getMessage(), e);
+            return "";
+        }
+    }
+
+    @Test
+    void verifyReportReader() {
+        String jsonFile = readFile("coverage-report.json");
+        IReportReader reportReader = new JsonReportReader();
+        Report report = reportReader.read(jsonFile);
+        assertEquals("org/example", report.getName());
+    }
+
+    @Test
+    void verifyInvalidJSON(){
+        IReportReader reportReader = new JsonReportReader();
+        try {
+            reportReader.read("%n");
+            fail();
+        }catch (ReportReaderException e){
+            assertEquals("Could not read coverage file", e.getMessage());
+        }
+    }
+}
```

Comment:
Don't forget to add the extra line.

Reply:
This method is used only for testing purposes. The HTTP client is fetching the JSON file as string.